### PR TITLE
Ilya/fix/activation date

### PR DESCRIPTION
**Pull request by EndlessLifeCX**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -18,5 +18,6 @@
     "python.testing.unittestEnabled": false,
     "python.testing.nosetestsEnabled": false,
     "python.testing.pytestEnabled": true,
-    "python.formatting.provider": "black"
+    "python.formatting.provider": "black",
+    "python.pythonPath": ".venv\\Scripts\\python.exe"
```
Comment (kipod) [inline .vscode/settings.json]: please revert this

File: app/controller/account.py [hunk 2]
```diff
@@ -50,12 +50,12 @@ def add_ninja_invoice(account: Account, is_new: bool, mode: str):
...
-    invoice_date = datetime.now()
-    if account.activation_date > invoice_date:
-        invoice_date = invoice_date.date().replace(day=1).strftime("%Y-%m-%d")
-    else:
-        invoice_date = (
-            account.activation_date.date().replace(day=1).strftime("%Y-%m-%d")
+    # invoice_date = datetime.now()
+    # if account.activation_date > invoice_date:
+    #     invoice_date = invoice_date.date().replace(day=1).strftime("%Y-%m-%d")
+    # else:
```
Comment (kipod) [inline app/controller/account.py]: don't need store old code in comments, we use git to store old code version

File: app/controller/account.py [hunk 3]
```diff
@@ -50,12 +50,12 @@ def add_ninja_invoice(account: Account, is_new: bool, mode: str):
             .filter(ResellerProduct.months == account.months)
             .first()
         )
-    invoice_date = datetime.now()
-    if account.activation_date > invoice_date:
-        invoice_date = invoice_date.date().replace(day=1).strftime("%Y-%m-%d")
-    else:
-        invoice_date = (
-            account.activation_date.date().replace(day=1).strftime("%Y-%m-%d")
+    # invoice_date = datetime.now()
```
Comment (viktor-radchenko) [inline app/controller/account.py]: Remove comments

File: app/js/main.js [hunk 4]
```diff
@@ -8,11 +9,50 @@ if (id.isNan){
...
+        urlParams.set(rows_param, rows_per_page)
+    }
+    window.open(`accounts?${urlParams}`,'_self')
+
+})
 
-$(document).ready(function() {
     $('#theTable').DataTable({
+        // "searching":false,
+        // "info" : false,
```
Comment (viktor-radchenko) [inline app/js/main.js]: Remove comments

File: app/ninja/invoice.py [hunk 5]
```diff
@@ -55,7 +55,7 @@ def delete_item(self, invoice_items):
         self.invoice_items = [
             data
             for data in self.invoice_items
-            if data["invoice_items"] != invoice_items
+            if data["notes"] != invoice_items['notes']
```
Comment (viktor-radchenko) [inline app/ninja/invoice.py]: Better use `data.get('notes')` and `invoice_items.get('notes')` to avoid crashes

File: app/forms/account.py [hunk 6]
```diff
@@ -20,5 +20,8 @@ class AccountForm(FlaskForm):
     activation_date = DateField(
         "Activation date", validators=[DataRequired()], default=datetime.now
     )
+    extension_date = DateField(
+        "Activation date", validators=[DataRequired()], default=datetime.now
```
Comment (viktor-radchenko) [inline app/forms/account.py]: Correct `Activation date` to `Extension date`

File: app/controller/account.py [hunk 7]
```diff
@@ -361,12 +362,54 @@ def save(self, form):
...
+                new_invoice_date = form.activation_date.data.replace(day=1).strftime(
+                    "%Y-%m-%d"
+                )
+                invoice_date = self.account.activation_date.replace(day=1).strftime(
+                    "%Y-%m-%d"
+                )
+                invoices = [i for i in NinjaInvoice.all() if not i.is_deleted]
+
+                if new_invoice_date != invoice_date:
+                    new_invoice = True
```
Comment (kipod) [inline app/controller/account.py]:
```python
is_new_invoice = (new_invoice_date != invoice_date)
```
and no need declare the variable  before

**Review by kipod — APPROVED** [review]

**Pull request merged**